### DISCUSSION TITLE
The session is not release when client.connect()  occur exception (e.getMessage() is null)

### DISCUSSION THREAD

**Discussion by zhengshubin** [eclipse-milo/milo#1105] [General]
**Describe the bug**
A clear and concise description of behavior being reported and, if applicable, why it's incorrect.
I had create OpcUaClient 
OpcUaClient client=OpcUaClient.create(...);

CompletableFuture<UaClient> completableFuture =client.connect();
And I had  catch like it:
CompletableFuture<CustomOpcUaClient> customFuture=new CompletableFuture<>();
completableFuture.whenComplete(
					(uaClient,e)->{
						if (e==null) {
							 customFuture.complete(new CustomOpcUaClient( client));
						}else {
							
							future.completeExceptionally(e);
						}
					}
			);
Actually,  e is not null,   I   try print  e.getMessage() but is null ,  And I must   connect  again  by scheduler, it was  same result.
But  after  many times    , it occur  UaServiceFaultException: status=Bad_TooManySessions.
So I think that  he session is not released.  even though exception occurred(e.getMessage()  is null) 

**Expected behavior**
A clear and concise description of what you expected to happen instead.

**Logs and Packet Captures**
If applicable, add application logs or Wireshark captures to help explain your problem.

**Additional context**
Version of Eclipse Milo you are using, OPC UA client or software server you are interacting with, etc...

**Comment by kevinherron** [comment]
You don’t need to keep calling `connect()`, if you call it once it will keep retrying.

Please fix your code example and provide code that reproduces the issue and a Wireshark capture if you continue to have a problem.
